src/commands/handlers/registration.js

```diff
@@ -437,7 +437,9 @@ function getSaslAuth(handler) {
         // An account username has been given, use it for SASL auth
         return {
             account: options.account.account,
-            password: options.account.password || '',
+            secret: (options.sasl_mechanism.toUpperCase() === 'AUTHCOOKIE' ?
```

Comment:
I'm not sure this is really belonging in that function.
At this point you might want to consider refactoring the SASL handling to the intended form, handling different mechs in their own functions, adding step functions etc.

But just adding a mech check in here seems strange to me.